llm/bill_on_document_created.py

```diff
@@ -0,0 +1,329 @@
+from firebase_functions.firestore_fn import (
+    Event,
+    DocumentSnapshot,
+)
+from llm_functions import get_summary_api_function, get_tags_api_function_v2
+from typing import TypedDict, NewType
+from collections import deque
+
+Category = NewType("Category", str)
```

Comment:
I am not sure how familiar people are with newtypes so I added some explainer in the readme file.

Reply:
The benefit is that you can't screw up topics versus categories. The classic meme is,

<img width="882" height="1200" alt="image" src="https://github.com/user-attachments/assets/b5d6e02a-c013-42bd-a76c-0bda5986d735" />

Yes, it is a slight effort to add the wrapper. But if you have functions which take category and topic you can't goof up the order in a non-obvious way.